PROTOCOL.md

```diff
@@ -288,11 +289,11 @@ Field    | Type | Description
 A message addressed to all subscriptions listening on the stream. It contains a `StreamMessage` as a payload at the Message Layer level. The `StreamMessage` representation is also an array (nested in the `BroadcastMessage` array) which is described in the [StreamMessage](#streammessage) section.
 
 ```
-[version, type, streamMessage]
+[version, type, requestId, streamMessage]
```

Comment:
Should the identifiers of responses be `responseId`?

Reply:
Or is the idea here that the `requestId` would be equivalent of the requestId of the original `SubscribeMessage`?

Reply:
Yes, in responses we echo back the `requestId` of the request that produced the response(s). That's why we're calling it `requestId` also in response messages.

Whether the `BroadcastMessage` needs it I'm not sure. It feels a bit redundant, but I put it there for completeness: otherwise it would be the only `ControlMessage` that doesn't contain a `requestId`.

Reply:
I think it would be useful indeed to have `BroadcastMessage` contain `requestId` as it could potentially help us diagnose issues with long running subscriptions. Care just needs to be taken in making sure that correct `requestId` is set for each client when broadcasting (it should match the `requestId` for that particular client-subscription).